enterprise/models.py

```diff
@@ -107,3 +110,56 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+def logo_path(instance, filename):
+    """
+    Delete the file if it already exist and returns the enterprise customer logo image path.
+
+    :param instance: EnterpriseCustomerBrandingConfiguration object
+    :param filename: file to upload
+    :return path: path of image file e.g. enterprise_logo/1/filename
+    """
+    name = os.path.join('enterprise_logo', str(instance.id), filename)
+    fullname = os.path.join(settings.MEDIA_ROOT, name)
+    if os.path.exists(fullname):
+        os.remove(fullname)
+    return name
+
+
+class EnterpriseCustomerBrandingConfiguration(TimeStampedModel):
+    """
+    Model that keeps track of enterprise branding configurations e.g. enterprise customer logo.
+
+    Fields:
+        enterprise_customer (ForeignKey[EnterpriseCustomer]): enterprise customer
+        logo (ImageField): enterprise customer image
+    """
+
+    enterprise_customer = models.OneToOneField(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        related_name='branding_configuration'
+    )
+    logo = models.ImageField(
+        upload_to=logo_path,
+        help_text=_(u'Please add only .PNG files for logo images.'),
+        null=True, blank=True, max_length=255,
+        validators=[validate_image_extension, validate_image_size]
+    )
+
+    class Meta:
+        """ Meta class for this Django model """
+        verbose_name = _('Branding Configuration')
+        verbose_name_plural = _('Branding Configurations')
+
+    def save(self, *args, **kwargs):
+        """save the enterprise customer branding config """
+        if self.pk is None:
+            logo_image = self.logo
+            self.logo = None
+            super(EnterpriseCustomerBrandingConfiguration, self).save(*args, **kwargs)
+            self.logo = logo_image
```

Comment:
@asadiqbal08 Why do you need to unset logo before calling base save (genuine question)?